target/linux/ramips/dts/MIWIFI-R3.dts

```diff
@@ -0,0 +1,180 @@
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "xiaomi,miwifi-r3", "ralink,mt7620a-soc";
+	model = "Xiaomi Mi Router R3";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio1 6 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		blue {
+			label = "miwifi-r3:blue:status";
+			gpios = <&gpio1 0 GPIO_ACTIVE_LOW>;
+		};
+
+		yellow {
+			label = "miwifi-r3:yellow:status";
+			gpios = <&gpio1 2 GPIO_ACTIVE_LOW>;
+		};
+
+		red {
+			label = "miwifi-r3:red:status";
+			gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	nand {
```

Comment:
It is 

```
nand: nand@800 {
	reg = <0x800 0x100>;

	pinctrl-names = "default";
	pinctrl-0 = <&nand_pins>;
}
```

Furthermore the whole nand node should be moved to mt7620a.dtsi with `status = "disabled"` and only referenced here via

```
&nand {
	status = "okay";
};
```

The nand pinmux should/need to be added to mt7620a.dtsi as well:

```
nand_pins: nand {
	nand {
		ralink,group = "nd_sd";
		ralink,function = "nand";
	};
};
```